### PR TITLE
feat: Initial version of Anthos Cluster blueprint

### PR DESCRIPTION
**Pull request by kaariger**
The Anthos Cluster blueprint creates an Anthos cluster with Anthos Config Management (ACM) installed.

Create an Anthos cluster using gke blueprint
Install ACM using the acm blueprint

### CODE REVIEW

File: catalog/anthos-cluster/acm/config-mgmt-iam.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+kind: IAMPartialPolicy
+metadata: # kpt-merge: platform-namespace/source-reader-sync-cluster-name-platform-project-id
+  name: source-reader-sync-cluster-name-platform-project-id # kpt-set: source-reader-sync-${cluster-name}-${platform-project-id}
+  namespace: platform-namespace # kpt-set: ${platform-namespace}
+  annotations:
+    cnrm.cloud.google.com/blueprint: cnrm/acm/v0.0.1
+    cnrm.cloud.google.com/project-id: platform-project-id # kpt-set: ${platform-project-id}
+spec:
+  resourceRef:
+    apiVersion: resourcemanager.cnrm.cloud.google.com/v1beta1
```
Comment (loudej) [inline catalog/anthos-cluster/acm/config-mgmt-iam.yaml:58]: Should be `sourcerepo.cnrm.cloud.google.com/v1beta1`?

Comment (kaariger) [reply catalog/anthos-cluster/acm/config-mgmt-iam.yaml:58]: Yes, this was fixed in https://github.com/GoogleCloudPlatform/blueprints/pull/88 but I still need to update the acm package in this blueprint.

File: catalog/anthos-cluster/Kptfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+kind: Kptfile
+metadata:
+  name: anthos-cluster
+  annotations:
+    blueprints.cloud.google.com/title: Anthos cluster blueprint
+upstream:
+  type: git
+  git:
+    repo: https://github.com/GoogleCloudPlatform/blueprints
+    directory: /catalog/gke
```
Comment (bharathkkb) [inline catalog/anthos-cluster/Kptfile:11]: If `anthos-cluster` is a top level blueprint in catalog, I think we should compose GKE under `/catalog/anthos-cluster/gke`. I will defer to @morgante if this is the recommended approach

File: catalog/anthos-cluster/Kptfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    ref: main
+    commit: f0d690ae0003f73aef68f885c0df5d2215e6beea
+info:
+  description: |
+    A GKE cluster with a primary node pool. An existing subnet needs to be provided where the cluster should be created.
+pipeline:
+  mutators:
+    - image: gcr.io/kpt-fn/apply-setters:v0.1
```
Comment (bharathkkb) [inline catalog/anthos-cluster/Kptfile:26]:
```suggestion
    - image: gcr.io/kpt-fn/apply-setters:v0.2
```

File: catalog/anthos-cluster/Kptfile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+upstreamLock:
+  type: git
+  git:
+    repo: https://github.com/GoogleCloudPlatform/blueprints
+    directory: /catalog/gke
+    ref: main
+    commit: f0d690ae0003f73aef68f885c0df5d2215e6beea
+info:
+  description: |
+    A GKE cluster with a primary node pool. An existing subnet needs to be provided where the cluster should be created.
```
Comment (bharathkkb) [inline catalog/anthos-cluster/Kptfile:23]: Lets update description

Comment (kaariger) [reply catalog/anthos-cluster/Kptfile]: updated

File: catalog/anthos-cluster/Kptfile (hunk 2 above)
Comment (kaariger) [reply catalog/anthos-cluster/Kptfile:11]: Based on a quick discussion, we want to keep these both as top level.

File: catalog/anthos-cluster/Kptfile [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  annotations:
+    blueprints.cloud.google.com/title: Anthos cluster blueprint
+upstream:
+  type: git
+  git:
+    repo: https://github.com/GoogleCloudPlatform/blueprints
+    directory: /catalog/gke
+    ref: main
+  updateStrategy: resource-merge
+upstreamLock:
```
Comment (bharathkkb) [inline catalog/anthos-cluster/Kptfile:14]: We should clean the upstream and upstream lock

File: catalog/anthos-cluster/setters.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: ConfigMap
+metadata: # kpt-merge: /setters
```
Comment (bharathkkb) [inline catalog/anthos-cluster/setters.yaml:17]:
```suggestion
metadata:
```

**Review by bharathkkb — CHANGES_REQUESTED** [review]

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**